### PR TITLE
fix: disable underlying data custom dimension

### PR DESCRIPTION
**Pull request by rephus**
Closes:  #7560

### Description:

Screenshot from 2023-10-27 15-37-58
Screenshot from 2023-10-31 08-37-08

Screenshot from 2023-10-31 08-37-17
Screenshot from 2023-10-31 08-38-11
Screenshot from 2023-10-31 08-43-18
Screenshot from 2023-10-31 08-43-58

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Review by ZeRego — APPROVED** [review]

**Review by ZeRego — CHANGES_REQUESTED** [review]
We also need to remove this option from other chart types.

**Comment by rephus** [comment]
> We also need to remove this option from other chart types.

I've been looking at the code, and I don't think we limit `view underlying data` based on the type (we don't have a single item/column like in the table (or table chart) , I think still makes sense to `view` based on the metric

> Could we rebase this branch once we merge https://github.com/lightdash/lightdash/pull/7677? does "view underlying data" work when results are pivoted by a bin?

It doesn't break, but it doesn't return the right results  (like you mention later) , mainly because we don't use the right filter / custom dimension on the `runUnderlyingDataQuery` method.

> Should we disable underlying data when a bin dimension is in the query? because when we see the underlying data of a metric we should add a filter "where the dimension used to bin has a value between X and Y" but that isn't happening.

Yep, you're correct, I guess this is the easiest solution for now.

File: packages/frontend/src/components/SimpleTable/CellContextMenu.tsx [hunk 1]
```diff
@@ -46,36 +53,40 @@ const CellContextMenu: FC<Pick<CellContextMenuProps, 'cell'>> = ({ cell }) => {
...
-                            });
-                        }}
-                    />
-                </Can>
-            )}
+            {item &&
+                !isDimension(item) &&
+                !isCustomDimension(item) &&
+                !hasCustomDimension(metricQuery) &&
+                !hasCustomDimension(metricQuery) && (
```
Comment (ZeRego) [inline packages/frontend/src/components/SimpleTable/CellContextMenu.tsx:60]: duplicate

File: packages/frontend/src/components/SimplePieChart/PieChartContextMenu.tsx [hunk 2]
```diff
@@ -156,10 +160,11 @@ const PieChartContextMenu: FC<PieChartContextMenuProps> = ({
                     Copy
                 </Menu.Item>
 
-                {canViewUnderlyingData ? (
+                {canViewUnderlyingData && !hasCustomDimension(metricQuery) ? (
                     <Menu.Item
                         icon={<MantineIcon icon={IconStack} />}
                         onClick={handleOpenUnderlyingDataModal}
+                        title="Underlying data is not available with custom dimensions."
```
Comment (ZeRego) [inline packages/frontend/src/components/SimplePieChart/PieChartContextMenu.tsx:167]: we don't need the title since is not visible

**Review by ZeRego — APPROVED** [review]
lgtm 👍  
Left a few suggestions.

File: packages/frontend/src/components/SimplePieChart/PieChartContextMenu.tsx (hunk 2 above)
Comment (rephus) [reply packages/frontend/src/components/SimplePieChart/PieChartContextMenu.tsx:167]: Yeah, sorry, a leftover from testing

**Comment by rephus** [comment]
Unrelated failling e2e tests

**Pull request merged**